pkgs/os-specific/linux/pam_rssh/default.nix

```diff
@@ -0,0 +1,48 @@
+{
+  lib,
+  fetchFromGitHub,
+  rustPlatform,
+  openssl,
+  pam,
+  pkg-config,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "pam_rssh";
+  version = "unstable-2023-01-12";
+
+  src = fetchFromGitHub {
+    owner = "z4yx";
+    repo = pname;
+    rev = "773823b9a8605436e5bcfabd1c3ff8deb8503f43";
+    hash = "sha512-rAr4ugo+IaxOKXtfEq9stK4edodork2uz10trHDv9MN4Oc296uDQjLh8Sle04Z4mbW56rWlT8eHrW6YpqGHFpw==";
+    fetchSubmodules = true;
+  };
+
+  cargoHash = "sha512-gSZ2EAhFJ1XqYmrVqJm3QP0l433XQxvkstkvyDAFZXB6KXiGzN0EoFxYoTdD0q2fE3tr5EBlmm4iC/Jo3wi4sg==";
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  buildInputs = [
+    openssl
+    pam
+  ];
+
+  doCheck = false;
+  # How do disable specific tests?
+  # checkFlags = [
+  #   # Expects $USER and $SSH_AUTH_SOCK
+  #   "--skip=lib::tests::sshagent_list_identities"
+  #   "--skip=lib::tests::sshagent_auth"
+  #   "--skip=lib::tests::sshagent_more_auth"
+  #   "--skip=lib::tests::parse_user_authorized_keys"
+  # ];
```

Comment:
```suggestion
  checkFlags = [
    # Expects $USER and $SSH_AUTH_SOCK
    "--skip=sshagent_list_identities"
    "--skip=sshagent_auth"
    "--skip=sshagent_more_auth"
    "--skip=parse_user_authorized_keys"
  ];
```
that does not work?

Reply:
No, I get the following error:

<details>
<summary>Details</summary>

```
error: environment variable `SSH_AUTH_SOCK` not defined
   --> src/lib.rs:198:42
    |
198 |         let mut agent = AgentClient::new(env!("SSH_AUTH_SOCK"));
    |                                          ^^^^^^^^^^^^^^^^^^^^^
    |
    = note: this error originates in the macro `env` (in Nightly builds, run with -Z macro-backtrace for more info)

error: environment variable `SSH_AUTH_SOCK` not defined
   --> src/lib.rs:213:42
    |
213 |         let mut agent = AgentClient::new(env!("SSH_AUTH_SOCK"));
    |                                          ^^^^^^^^^^^^^^^^^^^^^
    |
    = note: this error originates in the macro `env` (in Nightly builds, run with -Z macro-backtrace for more info)

error: environment variable `SSH_AUTH_SOCK` not defined
   --> src/lib.rs:235:42
    |
235 |         let mut agent = AgentClient::new(env!("SSH_AUTH_SOCK"));
    |                                          ^^^^^^^^^^^^^^^^^^^^^
    |
    = note: this error originates in the macro `env` (in Nightly builds, run with -Z macro-backtrace for more info)

error: environment variable `USER` not defined
   --> src/lib.rs:252:24
    |
252 |         let username = env!("USER");
    |                        ^^^^^^^^^^^^
    |
    = note: this error originates in the macro `env` (in Nightly builds, run with -Z macro-backtrace for more info)
```

</details>

Reply:
Can you prefixing the tests with ``test_``?